File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] sol = new int[2];
        int n = numbers.length;
        int i=0, j=n-1;

        while(i<j){
            if(numbers[i] + numbers[j] == target){
                sol[0] = i+1;
                sol[1] = j+1;
                return sol;
            }

            if(numbers[i]+numbers[j] < target)
                i++;
            else
                j--;
        }

        return sol;
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        left, right = 0, n-1

        while left < right:
            sum = nums[left] + nums[right]
            if sum == target: 
                return [left+1, right+1]
            elif sum > target:
                right -= 1
            else: 
                left += 1

        return [-1, -1]
        